### PR TITLE
Refactoring and documenting class

### PR DESCRIPTION
**Pull request by jeremyf**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Documentation Update

## Description

I came to this commit by looking for raw `where` method calls to the
Article object.  Originally, I was looking at `where.not(user_id: :id)`
but found this service class.

In this class, I saw some significant duplication and went with a bit of
refactoring and some documentation.

This refactoring removed some unnecessary guards and calculations (as
documented inline).

## Related Tickets & Documents

None.

## QA Instructions, Screenshots, Recordings

None.

### UI accessibility concerns?

None.

## Added/updated tests?

- [x] Yes; I mean the test is rather "Hey this results in valid SQL"

## [Forem core team only] How will this change be communicated?

- [x] This change does not need to be communicated, and this is why not: it's an isolated refactor and documentation update.

### CODE REVIEW

File: app/services/articles/suggest_stickies.rb [hunk 1]
```diff
@@ -23,31 +32,37 @@ def call
...
+    #       which meant we were running an unnecessary count query.
+    #       We also had a guard clause that said if we have more than
+    #       6 tag_articles, we should return an empty array.  However,
+    #       with the initial limit of 3 for tag articles, that guard
+    #       was unnecessary.
     def more_articles
-      return [] if tag_articles.size > 6
+      scope = Article
+        .where("comments_count > ?", comment_count_num)
+        .limit(7)
```
Comment (djuber) [inline app/services/articles/suggest_stickies.rb:51]: is this limit  7 = 10 - "initial limit of 3 for tag articles", or something else?

**Review by djuber — APPROVED** [review]

File: app/services/articles/suggest_stickies.rb (hunk 1 above)
Comment (jeremyf) [reply app/services/articles/suggest_stickies.rb:51]: It was 10 - initial limit of 3 (or less).  But also we then randomize later.  Almost went with a constant on this.

Comment (citizen428) [reply app/services/articles/suggest_stickies.rb:51]: I think I'd actually prefer a constant.

Comment (jeremyf) [reply app/services/articles/suggest_stickies.rb:51]: Named constant sounds good to me.

File: app/services/articles/suggest_stickies.rb [hunk 2]
```diff
@@ -1,7 +1,11 @@
 module Articles
   class SuggestStickies
+    # @todo Should we make this configurable in the admin interface?
```
Comment (citizen428) [inline app/services/articles/suggest_stickies.rb:3]: Almost certainly yes as these categories are quite DEV-specific.

**Review by citizen428 — APPROVED** [review]

**Pull request merged**